android-sdk/src/main/java/ai/evolv/android_sdk/EvolvClientImpl.java

```diff
@@ -203,36 +207,44 @@ public void invoke(Object value) {
+                    if (experiments.has("confirmations")) {
+                        confirmations = experiments.get("confirmations").getAsJsonArray();
+                    }
                 }
 
-                JsonObject confirmedCids = new JsonObject();
-                for (Map.Entry<String, JsonElement> entry : confirmations.entrySet() ) {
-                    confirmedCids = entry.getValue().getAsJsonObject().get("cid").getAsJsonObject();
+                JsonArray confirmedCids = new JsonArray();
+                for (JsonElement entry : confirmations) {
+                    confirmedCids.add(entry.getAsJsonObject().get("cid"));
                 }
 
-                JsonObject contaminations = new JsonObject();
-                if (remoteContext.has("contaminations")) {
-                    contaminations = remoteContext.get("contaminations").getAsJsonObject();
+                JsonArray contaminations = new JsonArray();
+
+                if (remoteContext.has("experiments")) {
+                    JsonObject experiments = remoteContext.get("experiments").getAsJsonObject();
+                    if (experiments.has("contaminations")) {
+                        contaminations = experiments.get("contaminations").getAsJsonArray();
+                    }
                 }
 
-                JsonObject contaminatedCids = new JsonObject();
-                for (Map.Entry<String, JsonElement> entry : contaminations.entrySet() ) {
-                    contaminatedCids = entry.getValue().getAsJsonObject().get("cid").getAsJsonObject();
+                JsonArray contaminatedCids = new JsonArray();
+                for (JsonElement entry : contaminations) {
+                    contaminatedCids.add(entry.getAsJsonObject().get("cid"));
                 }
 
                 JsonArray confirmableAllocations = new JsonArray();
-                for (Map.Entry<String, JsonElement> alloc : allocations.getAsJsonObject().get("allocations").getAsJsonObject().entrySet()) {
+                for (JsonElement alloc : allocations.getAsJsonArray()) {
 
-                    String cid = alloc.getValue().getAsJsonObject().get("cid").getAsString();
-                    String eid = alloc.getValue().getAsJsonObject().get("eid").getAsString();
+                    JsonObject allocObject = alloc.getAsJsonObject();
+                    String cid = allocObject.get("cid").getAsString();
+                    String eid = allocObject.get("eid").getAsString();
 
-                    if (!confirmedCids.has(cid)
-                            && !contaminatedCids.has(cid)
-                            && hasEntryPointEids(evolvStore.activeEids, eid)) {
-                        confirmableAllocations.add(alloc.getValue());
+                    if (!hasConfirmedCids(confirmedCids, cid)
+                            && !hasContaminatedCids(contaminatedCids, cid)
+                            && hasActiveEids(evolvStore.activeEids, eid)) {
```

Comment:
This still isn't filtering to only entry points

Reply:
fixed